pkg/ipmi/oem.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"unsafe"
+)
+
+const (
+	_IPMI_FB_OEM_NET_FUNCTION1 = 0x30
+
+	_FB_OEM_SET_BIOS_BOOT_ORDER = 0x52
+	_FB_OEM_GET_BIOS_BOOT_ORDER = 0x53
+)
+
+// Get BIOS boot order data and check if CMOS clear bit and valid bit are both set
+func (i *IPMI) IsCMOSClearSet() (bool, []byte, error) {
+	req := &req{}
+	req.msg.cmd = _FB_OEM_GET_BIOS_BOOT_ORDER
+	req.msg.netfn = _IPMI_FB_OEM_NET_FUNCTION1
+
+	recv, err := i.sendrecv(req)
+	if err != nil {
+		return false, nil, err
+	}
+	// recv[1] bit 1: CMOS clear, bit 7: valid bit, check if both are set
+	if len(recv) > 6 && (recv[1]&0x82) == 0x82 {
+		return true, recv[1:], nil
+	}
+	return false, nil, nil
+}
+
+// Set BIOS boot order with both CMOS clear and valid bits cleared
```

Comment:
what is this doing? I don't get it. Is this relevant for linuxboot?

Reply:
The main purpose of this feature is to reset BIOS configuration back to default, which is defined in the OEM command get/set BIOS boot order. Here the BIOS boot order is not the focus, we simply read them during get/set to match the command request data format.

Reply:
What is the nvram that you are clearing used for?

Reply:
I think right now the 114 bytes CMOS NVRAM is not used, but in case if data is also needed by other host FW component such as ME, it needs to be stored in CMOS.